File: Python/programmers/etc/level3/풍선_터트리기.py
def solution(a):
    def input_min(k):
        q = []
        t = k[0]
        for x in k:
            if t > x:
                t = x
            q.append(t)
        return q
    k = set(input_min(a) + input_min(a[::-1]))
    return len(k)


a = [9,-1,-5]
print(solution(a)) # 3

a = [-16,27,65,-2,58,-92,-71,-68,-61,-33]
print(solution(a)) # 6

File: Java/programmers/etc/level3/풍선터트리기.java
package CodeTest.Java.programmers.etc.level3;

import java.util.*;

public class 풍선터트리기 {
    public int solution(int[] a) {
        Set<Integer> set = new HashSet<>();
        int n = a.length;
        
        int _min = a[0];
        for (int i=0; i<n; i++) {
            _min = Math.min(_min, a[i]);
            set.add(_min);
        }

        _min = a[n-1];
        for (int i=n-1; i>=0; i--) {
            _min = Math.min(_min, a[i]);
            set.add(_min);
        }

        return set.size();
    }
}
